src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -193,96 +197,88 @@ public void doAddLinks() throws Exception
 		this.reportsAfterAddLinks(xrefReporter, duplicateIdentifierReporter, preAddLinksReport);
 		logger.info("Purging unused ReferenceDatabse objects.");
 		this.purgeUnusedRefDBs();
-
-		logger.info("Now checking links.");
-
-		String linksReport = this.checkLinks();
-		String diffReportName = LINK_CHECK_REPORTS_PATH + "/linkCheckSummaryReport" + DateTimeFormatter.ofPattern(DATE_PATTERN_FOR_FILENAMES).format(LocalDateTime.now()) + ".tsv";
-		Files.write(Paths.get(diffReportName), linksReport.getBytes() );
-
 		logger.info("Process complete.");
 	}
 
 	/**
-	 * Checks the links to external resources. Returns a
-	 * @return
+	 * @param linkCheckReportLines
+	 * @param linkCheckManager
+	 * @param refDBInst
+	 * @param reportLine
+	 * @param numLinkOK
+	 * @param numLinkNotOK
 	 */
-	private String checkLinks()
-	{
-		StringBuilder linkCheckReportLines = new StringBuilder("RefDBName\tNumOK\tNumNotOK\n");
-		// Now, check the links that were created to ensure that they are all valid.
-		LinkCheckManager linkCheckManager = new LinkCheckManager();
-		linkCheckManager.setDbAdaptor(this.dbAdapter);
-		// Filter by references database name.
-		for (GKInstance refDBInst : LinksToCheckCache.getCache().keySet() )
+	private String checkLinksForRefDB(LinkCheckManager linkCheckManager, GKInstance refDBInst)
+	{
+		 int numLinkOK = 0;
+		 int numLinkNotOK = 0;
+		logger.info("Link-checking for database: {}", refDBInst.getDisplayName());
+		StringBuilder reportLine = new StringBuilder();
+		if (!refDBInst.getDisplayName().toUpperCase().contains(ENSEMBL))
+		{
+			reportLine.append(refDBInst.getDisplayName()).append("\t");
+		}
+		else
+		{
+			try
+			{
+				Optional<String> longerName = ((List<String>)refDBInst.getAttributeValuesList(ReactomeJavaConstants.name)).stream().filter(name -> name.length() > ENSEMBL.length()).sorted().findFirst();
+				reportLine.append(longerName.orElse(ENSEMBL)).append("\t");
```

Comment:
The details of how the longest name is found are pretty involved.  Could this be abstracted to a named function so the flow of the method doesn't get interrupted trying to understand how the longest name is found?  Also, the logging on exception could be handled in that function as well if there is a problem.

Reply:
Not a fan of this refactoring.